test/minishift/test_suite.go

```diff
@@ -0,0 +1,161 @@
+package minishift
+
+import (
+	"context"
+	"crypto/tls"
+	"github.com/fabric8-services/fabric8-common/log"
+	"github.com/fabric8-services/fabric8-tenant/auth"
+	"github.com/fabric8-services/fabric8-tenant/cluster"
+	"github.com/fabric8-services/fabric8-tenant/configuration"
+	"github.com/fabric8-services/fabric8-tenant/environment"
+	"github.com/fabric8-services/fabric8-tenant/openshift"
+	"github.com/fabric8-services/fabric8-tenant/test"
+	"github.com/fabric8-services/fabric8-tenant/test/gormsupport"
+	"github.com/fabric8-services/fabric8-tenant/test/resource"
+	"github.com/fabric8-services/fabric8-tenant/test/stub"
+	"github.com/satori/go.uuid"
+	"github.com/stretchr/testify/assert"
+	"net/http"
+	"os"
+	"sync"
+	"testing"
+	"time"
+)
+
+// TestSuite is a base for tests using Minishift and gorm db
+type TestSuite struct {
+	gormsupport.DBTestSuite
+	ClusterService      *stub.ClusterService
+	AuthService         *stub.AuthService
+	Config              *configuration.Data
+	toReset             func()
+	minishiftUrl        string
+	minishiftAdminName  string
+	minishiftAdminToken string
+	minishiftUserName   string
+	minishiftUserToken  string
+}
+
+func (s *TestSuite) SetupTest() {
+	resource.Require(s.T(), resource.Database, "F8_MINISHIFT")
+	s.minishiftUrl = os.Getenv("F8_MINISHIFT_URL")
```

Comment:
this could be handled by a "minishift configuration" type managed by spf13/viper (similar to the regular service configuration)

Reply:
well, I was too lazy to create a special config implementation just because of a few variables that are used only in tests anyway. But I can do that :-)

Reply:
done in ebdd969